packages/frontend/src/components/NavBar/index.tsx

```diff
@@ -114,11 +113,12 @@ const NavBar = memo(() => {
                 <Box sx={{ flexGrow: 1 }} />
 
                 <Group sx={{ flexShrink: 0 }}>
+                    {errorLogs.length > 0
```

Comment:
This logic should be in the hook not in the rendering function.

Here it should just be:

```
errorLogs.length > 0 ? <render> : null
```

any logic related to changing the logs themselves (i.e. setting visible) should be in the hook

Reply:
Setting visible is the trigger to open the drawer and show the errors in the UI, it doesn't modify the logs... previously there was a button that triggered the opening of the drawer to show the errors, now we're doing it based on having unread logs or not.
I tried to store this "logic" in the hook which is a provider actually 😅  and it gave me a "Too many re-renders. React limits the number of renders to prevent an infinite loop." error.
I think it's fine if it stays here.